backend/api/Services/AreaService.cs

```diff
@@ -0,0 +1,215 @@
+
+        public abstract Task<Area?> ReadById(string id);
+
+        public abstract Task<IEnumerable<Area>> ReadByInstallation(string installationCode);
+
+        public abstract Task<Area?> ReadByInstallationAndName(string installationCode, string areaName);
+
+        public abstract Task<Area> Create(CreateAreaQuery newArea);
+
+        public abstract Task<Area> Create(CreateAreaQuery newArea, List<Pose> safePositions);
+
+        public abstract Task<Area> Update(Area area);
+
+        public abstract Task<Area?> AddSafePosition(string installationCode, string areaName, SafePosition safePosition);
+
+        public abstract Task<Area?> Delete(string id);
+
+    }
+
+    [System.Diagnostics.CodeAnalysis.SuppressMessage(
+        "Globalization",
+        "CA1309:Use ordinal StringComparison",
+        Justification = "EF Core refrains from translating string comparison overloads to SQL"
+    )]
+    [System.Diagnostics.CodeAnalysis.SuppressMessage(
+        "Globalization",
+        "CA1304:Specify CultureInfo",
+        Justification = "Entity framework does not support translating culture info to SQL calls"
+    )]
+    public class AreaService : IAreaService
+    {
+        private readonly FlotillaDbContext _context;
+        private readonly IInstallationService _installationService;
+        private readonly IPlantService _plantService;
+        private readonly IDeckService _deckService;
+
+        public AreaService(
+            FlotillaDbContext context, IInstallationService installationService, IPlantService plantService, IDeckService deckService)
+        {
+            _context = context;
+            _installationService = installationService;
+            _plantService = plantService;
+            _deckService = deckService;
+        }
+
+        public async Task<IEnumerable<Area>> ReadAll()
+        {
+            return await GetAreas().ToListAsync();
+        }
+
+        private IQueryable<Area> GetAreas()
+        {
+            return _context.Areas.Include(a => a.SafePositions)
+                .Include(a => a.Deck).Include(d => d.Plant).Include(i => i.Installation);
+        }
+
+        public async Task<Area?> ReadById(string id)
+        {
+            return await GetAreas()
+                .FirstOrDefaultAsync(a => a.Id.Equals(id));
+        }
+
+        public async Task<Area?> ReadByInstallationAndName(Installation? installation, string areaName)
+        {
+            if (installation == null)
+                return null;
+
+            return await _context.Areas.Where(a =>
+                a.Name.ToLower().Equals(areaName.ToLower()) &&
+                a.Installation.Id.Equals(installation.Id)
+            ).Include(a => a.SafePositions).Include(a => a.Installation)
+                .Include(a => a.Plant).Include(a => a.Deck).FirstOrDefaultAsync();
+        }
+
+        public async Task<Area?> ReadByInstallationAndName(string installationCode, string areaName)
```

Comment:
Both this and the previous have the same name ReadByInstallationAndName, I think this one should be ReadByInstallationCodeAndName, no?